apps/cowswap-frontend/src/modules/onchainTransactions/updaters/FinalizeTxUpdater/services/checkSafeTransaction.ts

```diff
@@ -0,0 +1,55 @@
+import { updateSafeTransaction } from 'legacy/state/enhancedTransactions/actions'
+import { EnhancedTransactionDetails } from 'legacy/state/enhancedTransactions/reducer'
+
+import { finalizeEthereumTransaction } from './finalizeEthereumTransaction'
+import { handleTransactionReplacement } from './handleTransactionReplacement'
+
+import { CheckEthereumTransactions } from '../types'
+
+export function checkSafeTransaction(transaction: EnhancedTransactionDetails, params: CheckEthereumTransactions) {
+  const { chainId, getTxSafeInfo, dispatch, safeInfo, getReceipt, lastBlockNumber } = params
+  const { hash, receipt } = transaction
+  // Get safe info and receipt
+  const { promise: safeTransactionPromise, cancel } = getTxSafeInfo(hash)
+
+  // Get safe info
+  safeTransactionPromise
+    .then(async (safeTransaction) => {
```

Comment:
Same nitpick about making this fn async/await rather than .then/.catch

Reply:
https://github.com/cowprotocol/cowswap/pull/4139#discussion_r1549050160